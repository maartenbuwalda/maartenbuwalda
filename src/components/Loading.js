import React  from 'react'
import styled from 'styled-components'

const LoadingIndicator = styled.div`
  width: 1rem;
  height: 1rem;
  background: blue;
`

const Loading = () => <LoadingIndicator />

export default Loading