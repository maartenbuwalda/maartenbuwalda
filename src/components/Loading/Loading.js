import React from 'react'
import styled from 'styled-components'
import colors from '../../constants/colors'

const LoadingIndicator = styled.div`
  width: 10rem;
  height: 10rem;
  border-radius: 100%;
  background: ${colors.darkGray};
  animation: indicating 1s infinite;


  @keyframes indicating {
    0% {
      opacity: 1;
      transform: scale(0);
    } 100% {
      transform: scale(1.0);
      opacity: 0;
    }
  }
`

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`

const Loading = () => (
  <Wrapper>
    <LoadingIndicator />
  </Wrapper>
)

export default Loading
