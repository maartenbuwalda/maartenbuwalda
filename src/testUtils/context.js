const context = {
  theme: {},
  updateTheme: jest.fn(),
}

export default context
