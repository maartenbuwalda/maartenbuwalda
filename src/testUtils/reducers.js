export const reducerReturnsInitialState = (reducer, initialState) => {
  it('should succeed: return the initialState', () => {
    expect(reducer(undefined, {})).toEqual(initialState)
  })
}
