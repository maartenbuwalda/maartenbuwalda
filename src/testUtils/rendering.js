import toJson from 'enzyme-to-json'
import { shallow, mount } from 'enzyme'
import configureStore from 'redux-mock-store'
import 'jest-styled-components'

export const componentRenders = component => {
  it('renders correctly', () => {
    const shallowed = shallow(component)
    expect(toJson(shallowed)).toMatchSnapshot()
  });
}

export const connectedComponentRenders = component => {
  const middlewares = []
  const mockStore = configureStore(middlewares)
  const initialState = {}
  const store = mockStore(initialState)
  
  it('renders correctly', () => {
    const shallowed = shallow(component, store)
    expect(toJson(shallowed)).toMatchSnapshot()
  })
}
