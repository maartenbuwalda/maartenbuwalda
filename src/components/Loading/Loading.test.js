import React from 'react'
import Loading from './index'
import { componentRenders } from '../../testUtils/rendering'

describe('Loading component', () => {
  const component = <Loading />
  componentRenders(component)
})
