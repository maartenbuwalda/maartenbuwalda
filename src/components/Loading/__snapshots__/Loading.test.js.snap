// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading component renders correctly 1`] = `
<styled.div>
  <styled.div />
</styled.div>
`;
